src/components/connection-map/ConnectionMap.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  ComposableMap,
+  Geographies,
+  Geography,
+  Line,
+  Marker,
+} from 'react-simple-maps';
+
+import { geoCentroid, geoDistance } from 'd3-geo';
+import { theme } from 'src/styles/constants/theme';
+import { ICountryOption } from 'src/types/ICountry';
+import { IGeoJsonWorld } from 'src/types/IGeoJsonWorld';
+import { getCountryCodeByName } from 'src/utils/getCountryCodeByName';
+import styled from 'styled-components';
+import { feature } from 'topojson-client';
+
+const Map = styled.div`
+  background: ${theme['gray-l80']};
+
+  * {
+    outline: none;
+  }
+`;
+
+const getScaleMultiple = ({
+  distance,
+  verticalDistance,
+}: {
+  distance: number;
+  verticalDistance: number;
+}) => {
+  if (distance < 0.5 && verticalDistance < 45) {
+    // US to MX (Mexico) = .36
+    return 700;
+  }
+  if (distance < 1 && verticalDistance < 50) {
+    // US to GL (Greenland) = .68
+    return 400;
+  }
+  if (distance < 1.5 && verticalDistance < 55) {
+    // US to CH (Switzerland) = 1.23
+    return 200;
+  }
+  if (distance < 1.75 && verticalDistance < 60) {
+    // US to KZ (Kazakhstan) = 1.509
+    return 150;
+  }
+  if (distance < 2 && verticalDistance < 65) {
+    // US to SA (Saudia Arabia) = 1.83
+    return 100;
+  }
+  if (distance < 2.5 && verticalDistance < 70) {
+    // US to AU (Australia) = 2.27
+    return 75;
+  }
+  // US to MG (Madagascar) = 2.52
+  return 50;
+};
```

Comment:
This is a pretty cool feature! It looks like this is setting the zoom scale for the map based on how far the connecting line has to stretch. Are the country comments just giving an example of the distance or does this only work for shipments going out of the US?

Reply:
The country comments are just examples.